### PR TITLE
KnotxServer response configuration - wildcards test, case-insensitive filtering allowed headers

### PR DESCRIPTION
**Pull request by dragarth-7**
## Description

For a reason we can use:
```
AllowedHeadersFilter.CaseInsensitive.create(strings);
```
I done refactor usage of AllowedHeadersFilter

I also found one thing that worries me. In ```EndpointOptions``` we can also set:
```
private JsonObject additionalHeaders;
private List<Pattern> allowedRequestHeadersPatterns;
```
by provided setter. But if we use only ```setAllowedRequestHeaderPatterns```, we don't fixing value of additionalHeaders. But in the other way, if we use ```setAllowedRequestHeaders``` we also set correct value for allowedRequestHeadersPatterns. In my opinion we should make setAllowedRequestHeaderPatterns private, or even skip this concept. @Skejven @tomaszmichalak  what is your opinion about that ?

### knotx-commons
[#7](https://github.com/Knotx/knotx-commons/issues/7) and [PR-8](https://github.com/Knotx/knotx-commons/pull/8)
### knotx-server-http
[#41](https://github.com/Knotx/knotx-server-http/issues/41)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

**Comment by Skejven** [comment]
Hi @dragarth-7 
Thanks for bringing this problem. Actually, there are several tickets/topic that bring allowed headers filtering topic:
- https://github.com/Knotx/knotx-server-http/issues/41
- https://github.com/Knotx/knotx/issues/414
- https://github.com/Knotx/knotx/issues/401

There is few things to remember about and why restricting headers is present:
1. We don't want to send all original request (the one that comes from the user to server) be exposed to all external endpoints Knot.x will interact with (e.g. security reasons).
2. The same goes the opposite direction, we don't want to expose all headers from endpoints to the end user with the final response headers.
3. Somehow managing headers should be done at each level, that means every time there is interaction with:
- external endpoint (http-repository, http-action)
- final response to end-user (response-writer)
The current solution remembers Knot.x 1.0 times, and it might be a good time to start discussion how we can refactor the code to make managing headers easier in multiple places.

> FYI `Knotx/knotx` is repository for discussing general (cross-repo) problems and questions.

**Comment by tomaszmichalak** [comment]
Please resolve conflicts.

**Comment by dragarth-7** [comment]
Done. Conflicts are resolved.  Please remember that you need firstly changes from https://github.com/Knotx/knotx-commons/pull/14

File: handler/actions/src/test/java/io/knotx/fragments/handler/action/http/HttpActionTest.java [hunk 1]
```diff
@@ -92,6 +92,7 @@
       "  \"url\": \"http://knotx.io\",\n" +
       "  \"label\": \"Product\"\n" +
       "}";
+  private static final Set<String> ALLOW_ALL_HEADERS = new HashSet<>(Arrays.asList(".*"));
```
Comment (Skejven) [inline handler/actions/src/test/java/io/knotx/fragments/handler/action/http/HttpActionTest.java]:
```suggestion
  private static final Set<String> ALLOW_ALL_HEADERS = Collections.singleton(".*");
```

Comment (dragarth-7) [reply handler/actions/src/test/java/io/knotx/fragments/handler/action/http/HttpActionTest.java]: Done

**Pull request merged**